template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerUserSyncRoleConfigProvider.java

```diff
@@ -39,19 +42,27 @@
     @Override
     public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         String adminGroup = virtualGroupService.getVirtualGroup(source.getVirtualGroupRequest(), UmsRight.RANGER_ADMIN.getRight());
-
+        String rangerAdminAsSysAdminConfigProvider = rangerAdminAsSysAdminConfigProvider(source);
         if (CloudPlatform.AZURE.equals(source.getCloudPlatform()) && source.getGeneralClusterConfigs().isEnableRangerRaz()
                 && source.getServicePrincipals() != null) {
             String servicePrincipals = getServicePrincipalsString(source);
             return List.of(
                     config(ROLE_SAFETY_VALVE, ConfigUtils.getSafetyValveProperty(RANGER_USERSYNC_UNIX_BACKEND, "nss")),
-                    config(ROLE_ASSIGNMENT_RULES, "&ROLE_SYS_ADMIN:g:" + adminGroup),
+                    config(ROLE_ASSIGNMENT_RULES, "&ROLE_SYS_ADMIN:g:" + adminGroup + rangerAdminAsSysAdminConfigProvider),
                     config(RANGER_USERSYNC_AZURE_MAPPING, servicePrincipals));
         } else {
             return List.of(
                     config(ROLE_SAFETY_VALVE, ConfigUtils.getSafetyValveProperty(RANGER_USERSYNC_UNIX_BACKEND, "nss")),
-                    config(ROLE_ASSIGNMENT_RULES, "&ROLE_SYS_ADMIN:g:" + adminGroup));
+                    config(ROLE_ASSIGNMENT_RULES, "&ROLE_SYS_ADMIN:g:" + adminGroup + rangerAdminAsSysAdminConfigProvider));
+        }
+    }
+
+    private String rangerAdminAsSysAdminConfigProvider(TemplatePreparationObject source) {
+        String cmVersion = getCmVersion(source);
+        if (isVersionNewerOrEqualThanLimited(cmVersion, CLOUDERAMANAGER_VERSION_7_2_8)) {
```

Comment:
Do we want the CM version or the CDH version here?

Reply:
CM version